src/mlpack/tests/random_forest_test.cpp

```diff
@@ -223,7 +223,7 @@ BOOST_AUTO_TEST_CASE(UnweightedCategoricalLearningTest)
 
   // Train a random forest and a decision tree.
   RandomForest<> rf(trainingData, di, trainingLabels, 5, 25 /* 25 trees */, 1,
-      1e-7, MultipleRandomDimensionSelect(4));
+      1e-7, 20, MultipleRandomDimensionSelect(4));
```

Comment:
I think it would be good if we can have a test which checks that the depth of the tree does not exceed actual maximum depth.

Reply:
I have added the test.